cg/src/domains/subjects/data-shares.service.js

```diff
@@ -25,6 +31,26 @@ class DataShareService {
         id: dataShareId
       });
   }
+
+  async getDataForShare(token) {
+    const [data] = await db('data_shares')
+      .where({ token })
+      .join('subjects', 'data_shares.subject_id', 'subjects.id')
+      .leftJoin('subject_keys', 'subjects.id', 'subject_keys.subject_id')
```

Comment:
Left Join means we want every entry that is on 'subjects' and also is is 'subject_keys' right?
If an entry is in 'subject_keys' but not in 'subjects' then it should be left out right